packages/aws-lambda-otel-extension/opt/otel-extension/internal/index.js

```diff
@@ -171,13 +171,17 @@ const responseHandler = async (span, { res, err }, isTimeout) => {
         const startTime = val.startTime || [0, 0];
         const endTime = val.endTime || [0, 0];
 
-        let attributes = val.attributes;
+        let attributes = {
+          ...val.attributes,
+          'sls.original_properties': Object.keys(val.attributes).join(','),
```

Comment:
This is added so that we can distinguish between which attributes come directly from open telemetry and which come come from our instrumentation 😎

Reply:
There are a few places in the UI where we want to only show open telemetry tags. More specifically this view in console 👇 
<img width="1624" alt="Screen Shot 2022-03-16 at 10 56 50 AM" src="https://user-images.githubusercontent.com/1753824/158632857-d916a9c6-6c14-48ee-a1f8-b87db28f8a2b.png">

Reply:
And why exactly it's just otel originated properties here? Isn't it more about those specific set of properties?

e.g. if otel would add `headers` is it ok for us, that it'll auto-magically appear in that section ?

Reply:
Sorry, for being picky. I'm also asking because goal is to have this package Serverless Console agnostic. We should also advertise as it as good to e.g. use with Datadog or other monitoring services.

I wonder if it's the right logical place for this logic

Reply:
Ok, and `source` on attribute value objects doesn't already answer that? Maybe we can distinguish it somehow that way, or add another property to those values?

It doesn't feel clean to create otel property, which is not about telemetry but about the categorization of otel properties.

Reply:
> Ok, and source on attribute value objects doesn't already answer that? Maybe we can distinguish it somehow that way, or add another property to those values?

Unfortunately no because that is not the shape of the object that is sent out to our APIs. And the trouble with adding new properties to the payload is that we are trying to keep this as otel compliant as possible so when we package up the trace/metrics payloads in the open telemeptry proto format so we cannot add properties to that payload since it has to match the proto format.

> It doesn't feel clean to create otel property, which is not about telemetry but about the categorization of otel properties.

I do see what you are saying but this is just another attribute that our platform needs. We are already using the attribute section to define other sls platform specific properties so it seems to make the most sense to add this along side the other sls specific properties.

Is the issue that we don't like where attribute is being aded in the code or that we don't want this attribute at all?

Reply:
I might have misunderstood what you meant by this 👇 

> Ok, and source on attribute value objects doesn't already answer that? Maybe we can distinguish it somehow that way, or add another property to those values?

Are you suggesting we should add this attribute in the `otel-payloads` file instead of adding it here?

Reply:
> Is the issue that we don't like where attribute is being aded in the code or that we don't want this attribute at all?

Issue is that we mix namespaces, and produce dirty dataset, by example following feels wrong:

```javascript
const personMeta = {
  properties: {
    name: "John",
    surname: "Doe",
    age: 37,
    nameRelatedPropertyNames: ["name", "surname"]
  }
}
```

It's quite problematic for any generic processor (which may not follow the meaning of each defined property, but would like to present them all). I believe way better to construct this would be as:

```javascript
const personMeta = {
  properties: {
    name: "John",
    surname: "Doe",
    age: 37,
  },
  nameRelatedPropertyNames: ["name", "surname"]
}
```

Still, question is, can we do something like that here?

Reply:
> Are you suggesting we should add this attribute in the otel-payloads f

if I see correctly, attributes are constructed as:

```javascript
[
  {
    key: 'faas.id',
    source: 'computeCustomArn',
    type: 'stringValue',
  },
  {
    key: 'faas.name',
    source: 'functionName',
    type: 'stringValue',
  }
]
```

So I was thinking that maybe if we'd prefix `source` of each of our property with `sls.` it'll give same effect. Or maybe we can add extra property e.g. `isSlsProperty: true` 

But I'm not yet that familiar with otel schema, not sure what can work best

Reply:
TBH I don't know where they have this posted online this has been one of the more frustrating things looking at the OTEL spec. 

To get this info I have been been referencing the typescript files on the trace library 👇 

```
node_modules/@opentelemetry/sdk-trace-base/build/src/export/ReadableSpan.d.ts
```

Reply:
The other place to reference would be the `trace.proto` file. This describes what the data must look like when we send it out to our API (Or any other otel supported API)

Reply:
I found it in code: https://github.com/open-telemetry/opentelemetry-js/blob/038bbdb67ad16d6fd8a26a3a81e42726ceccf04e/packages/opentelemetry-sdk-trace-base/src/export/ReadableSpan.ts

Still, it appears we already break that, at least I see we define properties not listed above, here:
https://github.com/serverless/runtime/blob/14631f0784528046f655b0030fed7a70635311a6/packages/aws-lambda-otel-extension/opt/otel-extension/internal/index.js#L184-L199

Do I miss something?

Reply:
It is [from here](https://github.com/open-telemetry/opentelemetry-proto). I just wasn’t sure how to pull that proto file into the project from that repo since I couldn’t find an npm package for it. 🤷‍♂️

Reply:
I see it's not an npm package, can we put comments in those files indicating where are they copied from?

Otherwise, let's take it in, eventually, when refactoring I'll think on whether we can improve that (it smells as extra technical debt now)

Reply:
Yeah that is a good call I can add that in this PR 👍

Also just for more context there is one more challenge with those files. We need to be able to keep those in sync with our backend ingestion app. The backend app is using the same proto files to decode payloads. We have not quite figured out how to keep both of these projects in sync. Just wanted to share this so you know that at some point we will need to figure that out as well 👍

Reply:
Great point @Danwakeem I'll take that into account